### DISCUSSION TITLE
How can I evaluate the switch T2EU1C-RF on a Sonoff RF (400Mhz) with a rule?

### DISCUSSION THREAD

**Discussion by ansim58** [arendst/Tasmota#21517] [Support request]
I have a T2EU1C button on a Sonoff RF. Tasmota 13.4.0 is running on the Sonoff-RF.
I want to create a rule that sends an html-get string to the IOBroker REST api when the button is pressed.
Does anyone have experience with this?
Thank you very much!

**Comment by sfromis** [comment]
If your setup is working, as in you getting `RfReceived` payload in the console, the docs example near the end of this page shows how you can have rules.
https://tasmota.github.io/docs/RF-Protocol/
And the `WebQuery` command can be used to make http requests (not https unless ESP32).

**Reply by ansim58** [reply]
Many thanks for the tip!
Unfortunately, I can't get 'RfReceived' on my MQTT server.
The template is probably not optimally configured.
I have the Sonoff RF2 template with GPIO0 as Button 1, GPIO12 as Relay 1 and GPOI13 as LED_i, nothing else. How can I change the template so that the RF signals are displayed?
Thank you very much!!!

**Reply by barbudor** [reply]
It was not suggested to process it on your server but using a rule in Tasmota to process it and call WebQuery commanf

**Reply by sfromis** [reply]
First, **did** you see the `RfReceived` payloads in the console (meaning Tasmota console)? If no, no go.

Note that if your Sonoff device is not a RF **bridge**, the RF processing is not available to Tasmota in any way, unless modifying the hardware.

**Reply by barbudor** [reply]
Ah, it's a Sonoff RF switch, not a Sonoff RF Bridge ?

Tasmota is not aware of the RF signal on that device as it is directly connected in hardware to the button
So pressing the RF remote will change the relay

You may build a rule based on Power status change to trigger your WebQuery

**Reply by ansim58** [reply]
No, there is no RFresult in the console. It only says: RESULT = {“POWER”: “OFF”}
So rf- communication only works with an external rf receiver? The Sonoff RF has a built-in receiver. It is a great pity if it can't be reached from Tasmota.
Thank you very much for your efforts!

**Reply by barbudor** [reply]
Actually I think you are confusing many things

On both T2EU1C-RF and Sonoff RF Switch, the RF is an INPUT to the device. There is no RF connections between both
If you want to trigger something when someone press on the T2EU1C-RF button, the rule has to be done on the T2EU1C-RF

**Comment by ansim58** [comment]
@barbudor: 
The scenario is as follows: The Sonoff-RF is controlled by IOBroker and switches a pump.
If the conditions are such that IOBroker switches the pump off, but I want to switch the pump on manually with the T2EU1C-RF, then the IOBroker switches the pump off again immediately. I therefore want to be able to distinguish whether the pump was switched on or off manually or by IOBroker.

**Reply by sfromis** [reply]
Only way of doing that would be to have your flow sending commands from IOBroker know that it was from there. Then, if it was not doing something, the reason would be "something else", like buttons on the device, RF control, rules, timers or whatever.

If you still want `WebQuery` from Tasmota, you could send another command like `Event` to have rules power on/off, and manage a state to control the send depending on that, more rules.

**Reply by sfromis** [reply]
Another approach I already hinted at could be to modify your hardware, and have the RF signal taken away from direct relay control, to instead be connected to the ESP chip where Tasmota is running, and then build/config Tasmota for RF.

**Reply by ansim58** [reply]
Thank you very much for that!
This would mean to replace the Sonoff RF with a ESP RF-Bridge?

**Reply by sfromis** [reply]
Not that way. What you could do with a bridge device would be to receive the RF signals, and have some automation to transmit commands to a device with a relay. That means an extra device, unless you modify your existing device to wire the RF to the ESP chip instead of the relay.

**Comment by ansim58** [comment]
So, thanks to your suggestions and tips, I bought a Sonoff RF Bridge. I have flashed it with Tasmota. 
The T2EU buttons did not have to be paired, they are recognized automatically. So I didn't have to make a RF Firmware-Update. The RF Bridge sends a string to the MQTT server, which is evaluated with blockly. This solution is simple and works super well!
Many thanks to everyone, you have helped me a lot!
